WebRoot/js/zimbraMail/core/ZmOperation.js

```diff
@@ -306,9 +306,9 @@ ZmOperation.defineOperation =
 function(baseId, overrides) {
 	var id = (overrides && overrides.id) || (baseId && baseId.id) || baseId || Dwt.getNextId();
 	var textKey = (overrides && overrides.textKey) || ZmOperation.getProp(baseId, "textKey");
-	var text = textKey && ZmMsg[textKey];
+	var text = textKey && ZmMsg[textKey] || textKey;
```

Comment:
`textKey` should always be a "key" for translation.
Set an formatted/translated text label in operation object and load it here. Please find an appropriate name; `i18nText`, `translatedText`, `formattedText`, or any other better name.
```suggestion
	var text = textKey && ZmMsg[textKey] || ZmOperation.getProp(baseId, "i18nText");
```

Reply:
It may be better to check new prop (custom message) first.
```
var text = ZmOperation.getProp(baseId, "i18nText") || (textKey && ZmMsg[textKey]);
```

Reply:
sure we can do this, just a small change would be to use translatedText  instead of i18nText

Reply:
Updated as per suggestion.